### PR TITLE
make docker integration optional in the docker image

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Enable an orchestrator-only mode by only enabling the docker integration if the docker socket is found in `/var/run/docker.sock` or the `DOCKER_HOST` environment variable is set:

- don't enable the docker integration from the Dockerfile
- modify `51-docker.sh` to only run if aforementioned conditions are met
- add a `59-defaults.sh` that touches an empty conf if none of the previous detections got results. The agent will run without AD (could be used in sidecar mode after disabling default checks)
- add a `agent-nodocker.yaml` k8s manifest to run in OpenShift restricted mode
- rework all other entrypoints for consistency

### Motivation

Allow to run gracefully on OpenShift restricted mode.

### CODE REVIEW

**Comment by mfpierre** [comment]
the PR also removes the `procdir` & `cgroups` volumes, is this expected?

**Comment by xvello** [comment]
Yes, this is the manifest for the restricted openshift mode. I'll move it to https://github.com/DataDog/datadog-agent/pull/1366 instead

File: Dockerfiles/agent/entrypoint/51-docker.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#   - we detect the DOCKER_HOST envvar, overriding the default socket location
+#     (in that case, we trust the user wants docker integration and don't check existence)
+#   - we find the docker socket at it's default location
+
+if [[ -z "${DOCKER_HOST}" && ! -e /var/run/docker.sock ]]; then
+    exit 0
+fi
+
+
+cd /etc/datadog-agent/
```
Comment (mfpierre) [inline Dockerfiles/agent/entrypoint/51-docker.sh]: I would prefer avoid `cd` in this kind of scripts and work with absolute paths even if it's more verbose it's lets prone to errors, WDYT?

**Review by mfpierre — APPROVED** [review]
🐳

**Pull request merged**